media-sound/carla/carla-2.5.2.ebuild

```diff
@@ -0,0 +1,102 @@
+# Copyright 1999-2022 Gentoo Authors
```

Comment:
Can this be kept a symlink to the 9999 ebuild?

Reply:
It might. However, this is for one not what ::gentoo does (and I assume that they have enough experience to have at least tried or discussed using this method) and on the other hand I personally had a few problems with it:

* Although a non-issue in this overlay, it makes stabilization and in general setting KEYWORDS differently for different versions impossible
* If need for patching certain versions arises (e.g. to backport a fix), a symlink can not be used unless the patch is required for _all_ versions (including -9999)
* I would try to stick to the workflow used in ::gentoo just for the sake of not having to think about whether I commit to ::gentoo or ::audio-overlay.
* When applying changes to the -9999 ebuild, one must never forget to do check if this requires a revbump for the versioned ebuilds. This is easier to forget if one only has to edit -9999 instead of each versioned ebuild.
* This approach only works if the dependency and IUSE specifications are _exactly_ the same for each version (which breaks as soon as upstream introduces a new or removes an old dependency)
* It doesn't bring a noticeable reduction in size on users' devices (especially with Portage now keeping the git history for auto-synced repos to a minimum by using shallow clones and the like). At least not compared to the size of the ::gentoo repo (which we depend on).

I know that most of these points will probably be never or almost never a issue in reality. However, I've already had to deal with backporting fixes to certain versions or introducing new IUSE in newer versions that weren't backwards compatible to older versions. I've used the symlink-based workflow before and it brought me more problems than it made things simpler. And I'd rather have no symlinks in the off-chance we might need it in the future (which is becoming more likely with languages like Go and Rust, which pin their dependencies, growing in popularity or languages which are known for having many dependencies per package in general). We could switch to a copy-based workflow for single packages if needed. However, for the sake of uniformity I'd rather have a single policy for the whole overlay instead of some packages violating a common workflow.

Still, it is your overlay, so if you insist I'll revert to the symlink based workflow. Just be aware of the possible implications.

Reply:
FYI the reason for using symlinks is that otherwise the git log/diff is useless. It's one of the IMHO few/main issues with how Gentoo/Portage/PMS handles packages. So I would always use symlinks to ensure everyone has an easy way to check what has changed, be it in a PR or when just looking through the git log.
Most of the points you mentioned you can do with symlinks, but there's of course a balance to be struck, you don't want 10 conditionals on versions in your ebuild, so sometimes a versioned one makes more sense (I've pretty much only done this with patches that are no longer needed for newer versions or the live ebuild. I've ran into few if any (can't remember any) issues using this approach).

Reply:
This is a sensible explanation. I've kept 2.2.0 as a copy, though, since it requires applying a patch to build. 2.5.2, however, is now a symlink to 9999, conforming to your workflow.